flink-libraries/flink-table/src/main/scala/org/apache/flink/api/scala/table/TableEnvironment.scala

```diff
@@ -72,5 +74,68 @@ class TableEnvironment {
      new ScalaBatchTranslator(config).translate[T](table.relNode)
   }
 
-}
+  /**
+   * Registers a DataSet under a unique name, so that it can be used in SQL queries.
+   * The fields of the DataSet type are used to name the Table fields.
+   * @param name the Table name
+   * @param dataset the DataSet to register
+   */
+  def registerDataSet[T](name: String, dataset: DataSet[T]): Unit = {
```

Comment:
Does it make sense to have an `AbstractTableEnvironment` for the common parts of the Java and the Scala `TableEnvironment`s?